### ISSUE TITLE
Allow for controlling quote style when exporting to YAML 

### ISSUE THREAD

**Issue by willramirezdev** [cue-lang/cue#2727]
**Is your feature request related to a problem? Please describe.**
This builds on the conversation in [1368](https://github.com/cue-lang/cue/issues/1368). 

When using CUE to export to YAML, there are times when controlling the quote style is required. In this [playground example](https://cuelang.org/play/?id=I550O3XvZbA#cue@export@yaml), the wildcard entry is always export wrapped in single quotes (`'*.exmaple.com'`). In our use case, there are downstream tools which require double quotes (`"*.example.com"`).

**Describe the solution you'd like**
Allow some way to control quote style. In [1368](https://github.com/cue-lang/cue/issues/1368) there were some discussions about controlling this at the file level, but even some sort of field-level controls would would in our case.

**Describe alternatives you've considered**
As far as I'm aware, there are no ways to control this using CUE itself. There are some recommendations in Slack to implement some sort of post processing: find and replace on magic strings, use another format. 

**Additional context**
N/A

**Comment by mvdan** [comment]
> In our use case, there are downstream tools which require double quotes (`"*.example.com"`).

Can you please expand on this a bit? If there's a program that takes in YAML, are you saying that it only accepts double quotes and not single quotes? I don't think I've heard of partial YAML support like that before.

In #1368, Marcel also suggests eporting as JSON - it always double quotes strings, and YAML is a superset of JSON, so presumably that could be enough to satisfy another program which only knows how to consume double-quoted YAML strings.

**Comment by willramirezdev** [comment]
@mvdan yeah, in our case the tool downstream is setup with a linter to only allow a specific sub-set of YAML. 

To be clear: I don't think CUE is doing anything "wrong", but I suspect there will continue to be edge cases as long as there's no control over quote style. #2720 is another example where the fix was editing a specific regex within CUE to enforce quotes.

 In our case we followed Marcel's advice and changed our tooling to export JSON, so we have a workaround for now.

**Comment by ryepesg** [comment]
I am having the same issue. My use case is a regex that OpenTelemetry interprets different when the quotes are removed from the regex used in https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/fc363988211d90ee2ae7c75eda6ce90c04bae868/processor/redactionprocessor/config.go#L25

**Comment by mvdan** [comment]
> In our case we followed Marcel's advice and changed our tooling to export JSON, so we have a workaround for now.

I would suggest this as the better approach for naive consumers/parsers on the other side which don't actually know how to parse YAML correctly. JSON is a much simpler syntax, so it fits that use case better, and it is a valid subset of YAML so it shouldn't break any valid YAML parsers either.

**Comment by mvdan** [comment]
I'm going to close this for now because both users participating in this issue were dealing with faulty YAML parsers that did not correctly implement the YAML spec. A workaround for both of them appears to be to export as JSON rather than YAML, which is still valid as all JSON is valid YAML, and the simpler syntax should be less prone to errors with faulty parsers such as regular expressions.

If there are any remaining bugs relating to exporting as YAML, please feel free to comment below or raise a new issue. But I'm not sure we should do anything to appease incomplete or buggy YAML parsers, especially when JSON seems like a straightforward workaround if the parser cannot be fixed.
